doc/source/examples/incompressible-flow/2d-taylor-couette-flow-nitsche/2d-taylor-couette-flow-nitsche.rst

```diff
@@ -0,0 +1,358 @@
+We simulate the same case as the regular Taylor-Couette flow where the inner cylinder rotates at a constant angular anti-clockwise velocity :math:`\omega`, 
+while the outer cylinder is fixed. 
+The following figure shows the geometry of this problem and the corresponding boundary conditions. 
+
+.. image:: images/taylor-couette.png
+    :alt: The geometry and boundary conditions
+    :align: center
+    :name: geometry
+
+Parameter file
+--------------
+
+We first establish the meshes used for the simulation. Using Nitsche immersed boundaries, two meshes are to be defined : the fluid mesh and the geometry mesh (i.e. the inner cylinder).
+
+Mesh
+~~~~~
+
+The ``mesh`` subsection specifies the computational grid:
+
+.. code-block:: text
+
+  subsection mesh
+    set type = dealii
+    set grid type = hyper_ball
+    set grid arguments = 0, 0 : 1 : true
+    set initial refinement = 3
+  end
+
+The ``type`` specifies the mesh format used. We use the ``hyper_shell`` mesh generated from the deal.II `GridGenerator <https://www.dealii.org/current/doxygen/deal.II/namespaceGridGenerator.html>`_ . This GridGenerator generates the mesh of the interstice between two cocentric cylinder. The arguments of this grid type are the position of center of the cylinders (``0, 0``), the inner cylinder radius (`0.25`), the outer cylinder radius (`1`) and the number of subdivision in the azimuthal direction (`4`). All arguments are separated by ``:``. We set ``colorize=true`` and this sets the boundary ID of the inner cylinder to ``0`` and of the outer cylinder to ``1``
+
+The last parameter specifies the ``initial refinement`` of the grid. Most deal.II grid generators contain a minimal number of cells. The *hyper_shell* mesh is made of four cells. Indicating an ``initial refinement=3`` implies that the initial mesh is refined 3 times. In 2D, each cell is divided by 4 per refinement. Consequently, the final grid is made of 256 cells.
+
+.. image:: images/background_mesh_taylor-couette.png
+    :alt: The fluid mesh
+    :align: center
+    :name: fluid
+
+The ``Nitsche`` subsection specifies the solid geometry embedded in the fluid domain. The Nitsche Immersed Boundary (IB) uses particles located at the 
+Gauss quadrature points of the immersed mesh to represent the immersed body. For a thorough explanation of this, we refer the reader to `step-70 <https://www.dealii.org/developer/doxygen/deal.II/step_70>` of deal.II.
+
+.. code-block:: text
+
+  subsection nitsche
+  set beta = 10
+  set number of solids = 1
+  subsection nitsche solid 0
+	  subsection mesh
+		  set type = dealii
+		  set grid type = hyper_ball
+		  set grid arguments = 0, 0 : 0.25 : true
+		  set initial refinement = 6
+	  end
+	  subsection solid velocity
+		  set Function expression = -y ; x
+	  end
+    set enable particles motion		= false
+  end
+  set calculate torques on solid = true
+  set verbosity = verbose
+end
```

Comment:
The last "end" does not appear int the text code-block in the html generated!
I think you need to add an indentation so that the last "end" is within the code-block.